src/vec.rs

```diff
@@ -568,6 +569,28 @@ where
         other.into_iter().for_each(|i| _ = self.append(i))
     }
 
+    /// Try Append a new member to the end of the `identified_vec`, if the `identified_vec` already contains the element a InsertionError will be returned.
+    ///
+    /// - Parameter item: The element to add to the `identified_vec`.
+    /// - Returns: Either a Ok() with a pair `(inserted, index)`, where `inserted` is a Boolean value indicating whether
+    ///   the operation added a new element, and `index` is the index of `item` in the resulting
+    ///   `identified_vec`. If the given ID or Value pre-exists within the collection the function call returns a InsertionFailure Error specifying the reason.
+    /// - Complexity: The operation is expected to perform O(1) copy, hash, and compare operations on
+    ///   the `ID` type, if it implements high-quality hashing.
+    #[inline]
+    pub fn try_append(&mut self, element: Element) -> Result<(bool, usize), InsertionFailure> {
+        let id = self.id(&element);
+        if self.index_of_id(&id).is_some() {
+            return Err(InsertionFailure::ElementWithSameIDFound);
```

Comment:
this is maybe more useful:
```rust
{
..
   #[error("Duplicate element with same id found: '{}'")]
    ElementWithSameIDFound(String),
...
}
```

and then instead of `if self.index_of_id(&id).is_some()` you do:

```rust
return Err(InsertionFailure::ElementWithSameIDFound(format!("{:?}", id)));
```

Reply:
I missed this particular comment but that is definitely something we want. I'll made adjustments and add it.